src/utils/version_helper.py

```diff
@@ -2,44 +2,50 @@
 
 
 class VersionHelper:
-    VERSION_NUMBER_PLUS_SIGN_REGEX = r"^(\d+)\.(\d+)\.(\d+)\+(\d+)$"
+    VERSION_REGEX = (
+        r'^'                  # Start of the string
```

Comment:
Comments should be avoided as much as possible as they slow down the read of the code. 

If the code itself is not explicit enough and "can't" be that explicit then sure use comments (https://blog.cleancoder.com/uncle-bob/2017/02/23/NecessaryComments.html)

These comments are basically teaching "REGEX" -> Each DEV has to know about REGEX  and there are a lot of Books and documentation on the internet about it.

A REGEX starts with `^` and end with a `$` this is part of REGEX definition. (As an example, to me it looks like commenting a `;` in the end of a java line, the comment is not wrong but meh... what is the value of this comment, if I program in JAVA I am supposed to know that each lines end with a `;`.

A bit same here: "non-capturing group" "captured group" this is REGEX notions. I would not add it.

Talking about accepted parts of the regex well why not.. but the regex defines it already.


TLDR; I would not comment this regex at all, I would document the wanted behaviour in the function using :
1) Either in the doc of the function (comment doc) with a set of examples of accepted values + the general format evtl
2) And/Or in the tests, remember, tests are the best doc, they are always up to date! (**not** comments)
(this is what you did: tests + doc -> enough IMHO)

(Up to you to change it or not)

Reply:
> Comments should be avoided as much as possible as they slow down the reading of the code.

I can't entirely agree, especially in this context. If I need to spend more than 10 seconds to understand one line of code, then having a comment is valuable. Regex is the perfect example of this. For example, I know that `d` means a digit, but understanding `r"^(\d+)\.(\d+)\.(\d+)\+(\d+)$"`  requires parsing all the brackets and symbols in my head. I would prefer having a one liner comment with an example in this case.

Reply:
I wanted to be verbose because of the unintuitive nature of regex, but I might have gone too far.
Thanks for the feedback! I cleaned up the comments a little bit.